appveyor/numpy.patch

```diff
@@ -0,0 +1,831 @@
+         if (f_sources or fmodule_sources) and fcompiler is None:
+-            raise DistutilsError("extension %r has Fortran sources " \
+-                  "but no Fortran compiler found" % (ext.name))
++            raise DistutilsError("extension %r has Fortran sources "
++                                 "but no Fortran compiler found" % (ext.name))
+         if ext.language in ['f77', 'f90'] and fcompiler is None:
+-            self.warn("extension %r has Fortran libraries " \
+-                  "but no Fortran linker found, using default linker" % (ext.name))
+-        if ext.language=='c++' and cxx_compiler is None:
+-            self.warn("extension %r has C++ libraries " \
+-                  "but no C++ linker found, using default linker" % (ext.name))
++            self.warn("extension %r has Fortran libraries "
++                      "but no Fortran linker found, using default linker" % (ext.name))
++        if ext.language == 'c++' and cxx_compiler is None:
++            self.warn("extension %r has C++ libraries "
++                      "but no C++ linker found, using default linker" % (ext.name))
+ 
+-        kws = {'depends':ext.depends}
++        kws = {'depends': ext.depends}
+         output_dir = self.build_temp
+ 
+         include_dirs = ext.include_dirs + get_numpy_include_dirs()
+@@ -391,7 +430,7 @@ class build_ext (old_build_ext):
+                     if f in existing_modules:
+                         continue
+                     t = os.path.join(module_build_dir, f)
+-                    if os.path.abspath(f)==os.path.abspath(t):
++                    if os.path.abspath(f) == os.path.abspath(t):
+                         continue
+                     if os.path.isfile(t):
+                         os.remove(t)
+@@ -410,24 +449,57 @@ class build_ext (old_build_ext):
+                                            extra_postargs=extra_postargs,
+                                            depends=ext.depends)
+ 
++        if f_objects:
++            lib_name = ext.name
++            # link the fortran sources with gfortran
++            fcompiler.link_shared_object(f_objects, lib_name + '_gfortran.dll',
++                                         output_dir=self.build_temp,
++                                         library_dirs=[
++                                             self.build_temp,
++                                             '.',
++                                         ] + self.library_dirs,
++                                         extra_postargs=self.dlls + [
++                                             '-Wl,--out-implib,' + lib_name + '.dll.a',
++                                             '-Wl,--output-def,' + lib_name + '_gfortran.def',
++                                             '-Wl,--export-all-symbols',
++                                             '-Wl,--enable-auto-import',
++                                         ],
++                                         debug=self.debug)
++
```

Comment:
Again, if there is fortran code, then we need to build it into a separate DLL and then generate a lib for msvc.